setup.py

```diff
@@ -1,55 +1,88 @@
 """
-Setup Module to setup Python Handlers for the cs3api4lab extension.
+cs3api4lab setup
 """
-import os
+import json
+from pathlib import Path
 
 from jupyter_packaging import (
-    create_cmdclass, install_npm, ensure_targets,
-    combine_commands, ensure_python, get_version,
+    create_cmdclass,
+    install_npm,
+    ensure_targets,
+    combine_commands,
+    skip_if_exists,
 )
 import setuptools
 
-HERE = os.path.abspath(os.path.dirname(__file__))
+HERE = Path(__file__).parent.resolve()
 
 # The name of the project
-name="cs3api4lab"
+name = "cs3api4lab"
 
-# Ensure a valid python version
-ensure_python(">=3.5")
+lab_path = HERE / name / "labextension"
 
-# Get our version
-version = get_version(os.path.join(name, "_version.py"))
+# Representative files that should exist after a successful build
+jstargets = [
+    str(lab_path / "package.json"),
+]
 
+package_data_spec = {name: ["*"]}
 
-with open("README.md", "r") as fh:
-    long_description = fh.read()
+labext_name = "@sciencemesh/cs3api4lab"
+
+data_files_spec = [
+    ("etc/jupyter/jupyter_notebook_config.d", "jupyter-config/jupyter_notebook_config.d", "cs3api4lab.json"),
+    ("etc/jupyter/jupyter_server_config.d", "jupyter-config/jupyter_server_config.d", "cs3api4lab.json"),
+]
+
+cmdclass = create_cmdclass(
+    "jsdeps", package_data_spec=package_data_spec, data_files_spec=data_files_spec
+)
+
+js_command = combine_commands(
+    install_npm(HERE, build_cmd="build:prod", npm=["jlpm"]),
+    ensure_targets(jstargets),
+)
+
+is_repo = (HERE / ".git").exists()
+if is_repo:
+    cmdclass["jsdeps"] = js_command
+else:
+    cmdclass["jsdeps"] = skip_if_exists(jstargets, js_command)
+
+long_description = (HERE / "README.md").read_text()
+
+# Get the package info from package.json
+pkg_json = json.loads((HERE / "package.json").read_bytes())
 
 setup_args = dict(
     name=name,
-    version=version,
-    url="https://github.com/sciencemesh/cs3api4lab",
-    author="CS3MESH4EOSC",
-    description="CS3APIs Jupyterlab contents connector",
-    long_description= long_description,
+    version=pkg_json["version"],
+    url=pkg_json["homepage"],
+    author=pkg_json["author"],
+    description=pkg_json["description"],
+    license=pkg_json["license"],
+    long_description=long_description,
     long_description_content_type="text/markdown",
+    cmdclass=cmdclass,
     packages=setuptools.find_packages(),
     install_requires=[
-        "jupyterlab~=2.0",
+        "jupyterlab==3.0.16",
```

Comment:
Does it need to be this specifically? If there's nothing particularly blocking, I would allow some margin, at least on patch version, maybe minor as well.